### ISSUE TITLE
Running `run-all apply` on a fresh AWS account results in multiple bucket creation prompts

### ISSUE THREAD

**Issue by velovix** [gruntwork-io/terragrunt#1596]
Imagine you're a new developer who just set up their own AWS root account and want to deploy your own copy of your company's cloud infrastructure. Being somewhat savvy to the way Terragrunt works, you navigate to the directory containing deployment configuration for the `dev` stage and run `terragrunt run-all apply`. Unfortunately, instead of the infrastructure being created, you get multiple overlapping prompts notifying you that the state bucket doesn't exist with offers to create it.

It could be argued that this is `run-all` working as expected. After all, this is a natural result of running `terragrunt apply` on each module in parallel. However, it's a bit of an unfortunate experience. It would be nice if there was a clean way to deploy an entire stack for the first time with a single command and `terragrunt run-all apply` is nearly there.

The way I've worked around this is to instruct new developers to deploy a single module with no dependencies (i.e. the VPC module) first, which will take care of creating the state bucket for you.

**Comment by lorengordon** [comment]
@velovix Have you tried passing [`--terragrunt-non-interactive`](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-non-interactive), or setting the env [`TF_INPUT=0`](https://www.terraform.io/docs/configuration/environment-variables.html#tf_input)?

**Comment by velovix** [comment]
@lorengordon Thank you for the suggestion! That does work reasonably well. Naturally, since multiple commands are trying to create the state bucket at once, you'll get a few errors like this:

```
ERRO[0003] Create S3 bucket with retry ... returned an error: OperationAborted: A conflicting conditional operation is currently in progress against this resource. Please try again.
	status code: 409, request id: ..., host id: .... Sleeping for 10s and will try again.  prefix=[...] 
```

However, since there's a retry mechanism, this doesn't seem to cause any problems and the deployment eventually completes.

I think it would be nice to have a more bespoke way to deploy multiple modules at once that smooths out some of these rough edges, but I'm pleased with this solution. Thanks again!

**Comment by lorengordon** [comment]
@velovix One option would be to use a single bucket for the backend within any given project. Within the backend config, use something like `key = "tfstate/${path_relative_to_include()}/terraform.tfstate"` to create a statefile per config.

Though perhaps you are saying that *is* what you are doing, and with `run-all` several configs are triggering the logic to create the *one* bucket all at the same time? I don't really have a workaround if so. Would really need to just be much more explicit about the workflow... i.e. create the backend, then run terragrunt. For terragrunt to manage that workflow more completely, I think this feature would need to be implemented... https://github.com/gruntwork-io/terragrunt/issues/924

**Comment by velovix** [comment]
@lorengordon That's right, I use one state bucket per stage for all modules in that stage. Like you're saying, all the modules are attempting to create that same bucket at the same time.
